applications/StatisticsApplication/custom_methods/spatial_methods.h

```diff
@@ -0,0 +1,883 @@
+            method, method_name, non_historical_nodal_norm_module, NodalNonHistoricalSpatialMethods)     \
+        ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_NORM_METHOD_PYTHON_INTERFACE(                            \
+            method, method_name, non_historical_condition_norm_module,                                   \
+            ConditionNonHistoricalSpatialMethods)                                                        \
+        ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_NORM_METHOD_PYTHON_INTERFACE(                            \
+            method, method_name, non_historical_element_norm_module, ElementNonHistoricalSpatialMethods) \
+    }
+#endif
+
+namespace Kratos
+{
+///@addtogroup RANSApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+namespace SpatialMethods
+{
+template <typename TContainerType, typename TContainerItemType, template <typename T> typename TDataRetrievalFunctor>
+class ContainerSpatialMethods
+{
+public:
+    // special overloaded method for flags
+    int static CalculateSum(const ModelPart& rModelPart, const Flags& rVariable)
+    {
+        const TContainerType& r_container =
+            MethodUtilities::GetLocalDataContainer<TContainerType>(rModelPart);
+        int sum = 0;
+
+#pragma omp parallel for reduction(+ : sum)
+        for (int i = 0; i < static_cast<int>(r_container.size()); ++i)
+        {
+            const TContainerItemType& r_item = *(r_container.begin() + i);
+            sum += r_item.Is(rVariable);
+        }
+
+        sum = rModelPart.GetCommunicator().GetDataCommunicator().SumAll(sum);
+        return sum;
+    }
+
+    template <typename TDataType>
+    TDataType static CalculateSum(const ModelPart& rModelPart,
+                                  const Variable<TDataType>& rVariable)
+    {
+        KRATOS_TRY
+
+        const TContainerType& r_container =
+            MethodUtilities::GetLocalDataContainer<TContainerType>(rModelPart);
+
+        TDataType global_sum = rVariable.Zero();
+
+        if (r_container.size() > 0)
+        {
+            const TDataType& r_initial_value =
+                TDataRetrievalFunctor<TContainerItemType>()(*r_container.begin(), rVariable);
+
+            MethodUtilities::DataTypeSizeInitializer(global_sum, r_initial_value);
+
+#pragma omp parallel
+            {
+                TDataType sum = rVariable.Zero();
+                MethodUtilities::DataTypeSizeInitializer(sum, r_initial_value);
+
+#pragma omp for
+                for (int i = 0; i < static_cast<int>(r_container.size()); ++i)
+                {
+                    const TContainerItemType& r_item = *(r_container.begin() + i);
+                    const TDataType& current_value =
+                        TDataRetrievalFunctor<TContainerItemType>()(r_item, rVariable);
+                    MethodUtilities::DataTypeSizeChecker(current_value, sum);
+                    sum += current_value;
+                }
+#pragma omp critical
```

Comment:
asaik `atomic` is faster for such as simple case

Reply:
done :)